gui/mozregui/report_delegate.py

```diff
@@ -0,0 +1,51 @@
+from PyQt4.QtGui import QStyledItemDelegate, QStyleOptionProgressBarV2, \
+    QApplication, QStyle
+from PyQt4.QtCore import Qt, QRect, pyqtSlot as Slot
+
+
+class ReportItemDelegate(QStyledItemDelegate):
+    def __init__(self, parent=None, *args):
+        QStyledItemDelegate.__init__(self, parent, *args)
+        self.progress = 0
+        self.lastIndex = None
+
+    @Slot(object, int, int)
+    def show_dl_progress(self, dl, current, total):
+        self.progress = (current * 100) / total
+        if self.lastIndex:
+            self.lastIndex.model().dataChanged.emit(
+                self.lastIndex, self.lastIndex)
+
+    def paint(self, painter, option, index):
+        # Write status
+        QStyledItemDelegate.paint(self, painter, option, index)
+
+        if not index.row() == (index.model().rowCount() - 1):
+            return
+
+        # Check if download is on
+        if "Downloading" in str(index.data().toPyObject()):
```

Comment:
Same as above. Note that maybe others signals could help to detect start/end of downloading: https://github.com/mozilla/mozregression/blob/master/gui/mozregui/bisection.py#L20